bazel/BUILD

```diff
@@ -30,4 +33,28 @@ sh_binary(
         "@unused_deps//file",
         "@unused_deps_osx//file",
     ],
-)
\ No newline at end of file
+)
+
+# Toolchain that redefines singlejar to a version
+# that produces IntelliJ-compatible source JARs
+default_java_toolchain(
+    name = "java-toolchain",
+    source_version = "8",
+    target_version = "8",
+    singlejar = select({
+        "@bazel_tools//src/conditions:remote": ["@bazel_tools//src/tools/singlejar:singlejar"],
```

Comment:
I guess we will need to update `@bazel_tools//src/conditions:remote` with our own singlejar as well?

Reply:
@lolski Not necessarily. Currently we only indent to resolve the issue of "Go to source" not working in IntelliJ — only local machines are affected.

Reply:
@vmax is it as easy as this
```
"@bazel_tools//src/conditions:remote": [":singlejar_deploy.jar"]
```

or will it require more effort?

Ideally, we keep an identical implementation to avoid "bugs that only happen in local but not in RBE" and vice versa

Reply:
@lolski actually, separate `singlejar` implementations are already used in stock config

```
vmax@ghost build-tools java-toolchain-with-modified-singlejar bazel query --output=build @bazel_tools//tools/jdk:singlejar                         
DEBUG: Rule 'io_bazel_rules_python' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1545256788 -0500"
# /private/var/tmp/_bazel_vmax/6866c2e533a140c6a78f612a435b4e8b/external/bazel_tools/tools/jdk/BUILD:145:1
filegroup(
  name = "singlejar",
  srcs = select({"@bazel_tools//src/conditions:remote": ["@bazel_tools//src/tools/singlejar:singlejar"], "//conditions:default": ["@bazel_tools//tools/jdk:singlejar/singlejar"]}),
)
```

I don't know the *exact* reason for this, but RBE will continue to use the same implementation as previously; we only replace locally-used `singlejar` — which would be relatively safe

Reply:
ok, let's keep this in mind if we do encounter a weird JAR bug happening either only locally or in RBE.